src/vs/workbench/services/configurationResolver/common/variableResolver.ts

```diff
@@ -121,7 +125,7 @@ export class AbstractVariableResolverService implements IConfigurationResolverSe
 		}
 	}
 
-	private recursiveResolve(environment: IProcessEnvironment | undefined, folderUri: uri | undefined, value: any, commandValueMapping?: IStringDictionary<string>, resolvedVariables?: Map<string, string>): any {
+	private recursiveResolve(environment: IProcessEnvironment | IProcessEnvironment & { userHome: URI } | undefined, folderUri: uri | undefined, value: any, commandValueMapping?: IStringDictionary<string>, resolvedVariables?: Map<string, string>): any {
```

Comment:
Let's have a new type:

```
type Environment = { env?: IProcessEnvironment; userHome?: uri };
```

Then the first parameter of `recursiveResolve` is an `Environment`. 

Anywhere that we `await this._envVariablesPromise`, we can also `(await this._pathService.userHome())`.

Reply:
Okay, that seems to be working. Anywhere, in particular, I should define the type?

Reply:
Somewhere near the top of the file but after the imports is good.